pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/PrestoDateTruncFunction.java

```diff
@@ -0,0 +1,262 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import org.apache.pinot.core.common.DataSource;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.joda.time.Chronology;
+import org.joda.time.DateTimeField;
+import org.joda.time.DateTimeFieldType;
+import org.joda.time.DateTimeZone;
+import org.joda.time.DurationField;
+import org.joda.time.DurationFieldType;
+import org.joda.time.chrono.ISOChronology;
+import org.joda.time.field.DividedDateTimeField;
+import org.joda.time.field.OffsetDateTimeField;
+import org.joda.time.field.ScaledDurationField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Locale.ENGLISH;
+import static org.apache.pinot.core.operator.transform.function.TimeZoneKey.MAX_TIME_ZONE_KEY;
+import static org.apache.pinot.core.operator.transform.function.TimeZoneKey.getTimeZoneKeys;
+
+/**
+ * The <code>PrestoDateTruncFunction</code> class implements the sql compatible date_trunc function for TIMESTAMP type.
```

Comment:
Can you please add proper indentation to this javadoc?

Reply:
Sorry, I didn't follow. I followed the instructions given here: https://pinot.readthedocs.io/en/latest/dev_env.html. 

Can you let me know if the new indentation matches the requirement. Thanks. It passed checkstyle etc.

Reply:
I was referring to the indentation of the javadoc. Right now it is a little bit hard to read. You can check the javadoc in SelectionOperatorService.java as a reference.